packages/db/utils.js

```diff
@@ -6,17 +6,39 @@
  */
 export function normalizeUpload (upload) {
   const nUpload = { ...upload }
+  const backupUrls = nUpload.backupUrls ?? []
+  delete nUpload.backupUrls
   delete nUpload.content
   delete nUpload.sourceCid
 
+  const partOf = [...carUrlsFromBackupUrls(backupUrls)]
+
   return {
     ...nUpload,
     ...upload.content,
     cid: upload.sourceCid, // Overwrite cid to source cid
     pins: normalizePins(upload.content.pins, {
       isOkStatuses: true
-    })
+    }),
+    partOf
+  }
+}
+
+/**
+ * given array of backup_urls from uploads table, return a set of ipfs:// URIs for any CAR files in the backup_urls
+ * @param {string[]} backupUrls
+ * @returns {Iterable<string>}
+ */
+function carUrlsFromBackupUrls (backupUrls) {
```

Comment:
You probably want something like this so you can read CAR CID from S3 URLs also:

https://github.com/web3-storage/content-claims/blob/50932114d51cfe34da113ab065c7b35927dbc6c5/packages/lambda/src/lib/store/block-index.js#L171-L192

Reply:
done!